### PR TITLE
Emit console termination events

### PR DESCRIPTION
**Pull request by ttamimi**
With this change, the workloads controller will publish ConsoleTerminatedEvent
messages to Pub/Sub. The event will include per-container messages capturing
things like exit code, OS signal received, termination reason, and output
message (if any of these are available).

### CODE REVIEW

File: apis/workloads/v1alpha1/lifecycle_recorder.go [hunk 1]
```diff
@@ -208,3 +221,38 @@ func (l *lifecycleEventRecorderImpl) ConsoleTerminate(ctx context.Context, csl *
 	l.logger.Info("event recorded", "id", id, "event", events.EventTerminated)
 	return nil
 }
+
+func appendStatusMessages(result map[string]string, containers []corev1.ContainerStatus) {
```
Comment (theobarberbany) [inline apis/workloads/v1alpha1/lifecycle_recorder.go:225]:
```suggestion
func appendStatusMessages(result map[string]string, containerStatuses []corev1.ContainerStatus) {
```

Nit: Maybe this fits more naturally?

File: apis/workloads/v1alpha1/lifecycle_recorder.go [hunk 2]
```diff
@@ -208,3 +221,38 @@ func (l *lifecycleEventRecorderImpl) ConsoleTerminate(ctx context.Context, csl *
...
+	if containers == nil {
+		return
+	}
+
+	for _, containerStatus := range containers {
+		if containerStatus.State.Terminated != nil {
+			s := containerStatus.State.Terminated
```
Comment (theobarberbany) [inline apis/workloads/v1alpha1/lifecycle_recorder.go:232]: What is the state of a container that is in crashback loopoff? Will it be caught by this? Or is it handled by the waiting state below?

Comment (theobarberbany) [reply apis/workloads/v1alpha1/lifecycle_recorder.go:232]: I've just checked: looks like it'll be caught by waiting below

**Review by theobarberbany — APPROVED** [review]

**Comment by theobarberbany** [comment]
I think this looks good! Let's test it out :)

File: apis/workloads/v1alpha1/lifecycle_recorder.go (hunk 1 above)
Comment (ttamimi) [reply apis/workloads/v1alpha1/lifecycle_recorder.go:225]: Yes totally! I will change it.

File: apis/workloads/v1alpha1/lifecycle_recorder.go (hunk 2 above)
Comment (ttamimi) [reply apis/workloads/v1alpha1/lifecycle_recorder.go:232]: Yes here is a live example that was generated in lab:
https://gocardless.atlassian.net/browse/TCSR-290
The app container is in waiting state.

Comment (ttamimi) [reply apis/workloads/v1alpha1/lifecycle_recorder.go:232]: Well, not exactly a crashbackoff loop. It's an imagepullbackoff loop.

Comment (ttamimi) [reply apis/workloads/v1alpha1/lifecycle_recorder.go:232]: Nice! Thanks for checking 😊

**Review by theobarberbany — APPROVED** [review]

**Pull request merged**